src/Database/Schema/MysqlSchema.php

```diff
@@ -361,26 +362,41 @@ public function columnSql(TableSchema $schema, $name)
                     break;
             }
         }
-        $hasLength = ['integer', 'smallinteger', 'tinyinteger', 'string'];
+        $hasLength = [
```

Comment:
May be we re-order them like https://github.com/cakephp/cakephp/pull/10668/files#diff-e7a8f6f7caf1af99989fbae47509a4f8R382

```` php
        $hasLength = [
            TableSchema::TYPE_TINYINTEGER,
            TableSchema::TYPE_SMALLINTEGER,
            TableSchema::TYPE_INTEGER,
            TableSchema::TYPE_STRING
        ];
````

Or does it need to be ordered that way?